Sources/TuistSupport/Utils/Environment.swift

```diff
@@ -68,7 +68,8 @@ public class Environment: Environmenting {
     init(directory: AbsolutePath, fileHandler: FileHandling) {
         self.directory = directory
         self.fileHandler = fileHandler
-        setup()
+        #warning("This is causing a crash because `fileHandler` logs which uses `Environment.shared`")
+        // setup()
```

Comment:
This issue can be a separate PR but:

This is a chicken-egg problem, it was working previously but curious to see what we should do here, maybe revert?

Reply:
The issue:

- Using `Environment.shared` requires `Environment.init` to be called.
- `Environment.init` calles `setup` which uses `fileHandler`
- `FileHandler` logs which uses `Environment.shared` internally to determine when to color output, etc.

Hence the 💥

Reply:
Ah: https://github.com/tuist/tuist/pull/3752 (:

Reply:
I'm finishing that one up. I'll let you know so that you can rebase afterwards.